### PR TITLE
Added more docs for okta-auth-js

### PR DESCRIPTION
**Pull request by lboyette-okta**
@mystiberry-okta @joelfranusic-okta @raphaellondner-okta @slee-okta @karlmcguinness-okta @rchild-okta @larsjohansen-okta @stsai-okta 

This will serve as our docs for [the okta-auth-js sdk](https://github.com/okta/okta-auth-js).

### CODE REVIEW

File: _source/docs/guides/okta_auth_sdk.md [hunk 1]
```diff
@@ -4,138 +4,171 @@ title: Okta Auth SDK
...
 ---
 
-> This SDK is currently in **Beta** status.
-
 # Introduction
 
-The Okta Sign-In SDK builds on top of our [Authentication APIs](), allowing 
-you to create a custom login experience using JavaScript. This includes 
-a traditional Okta authentication flow with MFA or Social Auth.
+The Okta Auth SDK builds on top of our [Authentication API](/docs/api/resources/authn.html); enabling you to create a custom login experience using JavaScript.
```
Comment (rchild-okta) [inline _source/docs/guides/okta_auth_sdk.md]: "to enable" instead of ";enabling"

File: _source/docs/guides/okta_auth_sdk.md [hunk 2]
```diff
@@ -4,138 +4,171 @@ title: Okta Auth SDK
...
 
 # Prerequisites
 
-- An Okta org - Register for [Okta Developer Edition](https://www.okta.com/developer/signup/) if you don't have an existing org
-- [CORS enabled for your domain](/docs/api/getting_started/enabling_cors.html)
+- Use an existing Okta org or register for [Okta Developer Edition](https://www.okta.com/developer/signup/)
```
Comment (rchild-okta) [inline _source/docs/guides/okta_auth_sdk.md]: For a prereq, seems to make more sense to say "Have an existing ..." vs. "Use an existing"

Comment (rchild-okta) [inline _source/docs/guides/okta_auth_sdk.md]: Weird, in general I like the version before this change - was there a reason to switch?

File: _source/docs/guides/okta_auth_sdk.md [hunk 3]
```diff
@@ -4,138 +4,171 @@ title: Okta Auth SDK
...
+- [Add CORS to your domain in Okta](/docs/api/getting_started/enabling_cors.html)
 
 # Installation
 
 Include the following script in your page:
```
Comment (rchild-okta) [inline _source/docs/guides/okta_auth_sdk.md]: nit: "script tag" instead of "script"

File: _source/docs/guides/okta_auth_sdk.md [hunk 4]
```diff
@@ -4,138 +4,171 @@ title: Okta Auth SDK
...
-existing MFA. To track your progression through these states, we use a
-transaction. The possible states, their definitions, and their flow can be found
-in our [Authentication API documentation](/docs/api/resources/authn.html#transaction-state).
-
-Here is a list of the states and their possible methods:
-
-> Methods listed for certain states may only be available in certain scenarios
-
-> `stateToken` is never required as a parameter
+Providing only a username and password may not be sufficient to obtain a cookie. You may need to reset a password, enroll in multifactor authentication (MFA), or verify your existing MFA. To track your progression through these states, we use a transaction. The possible states, their definitions, and their flow can be found in our [Authentication API documentation](/docs/api/resources/authn.html#transaction-state). The following are collections of transaction states and their possible methods.
```
Comment (rchild-okta) [inline _source/docs/guides/okta_auth_sdk.md]: - "obtain a **session** cookie". 
- "reset **your** password"
- "we use a transaction **object**"

"The possible states..." -> "More information about the possible states, definitions, and flows can be found in our"

File: _source/docs/guides/okta_auth_sdk.md [hunk 5]
```diff
@@ -4,138 +4,171 @@ title: Okta Auth SDK
...
 
-> All transactions have:
```
Comment (rchild-okta) [inline _source/docs/guides/okta_auth_sdk.md]: Is a block quote how we get the highlighting on dev.okta.com?

File: _source/docs/guides/okta_auth_sdk.md (hunk 3 above)
Comment (mystiberry-okta) [inline _source/docs/guides/okta_auth_sdk.md]: Do we mean login (authentication) page? Not sure if everyone will be oriented this early in the doc.

File: _source/docs/guides/okta_auth_sdk.md (hunk 4 above)
Comment (mystiberry-okta) [inline _source/docs/guides/okta_auth_sdk.md]: 1. Can we say "To track your progression, we use transaction states." to make clear what the transaction is.
2. It's not super clear what's in this doc vs. auth.html. Can that last sentence make it clearer? Something like "The following transaction states and their methods are those most commonly used in an authentication flow" or something. Then make that 4th sentence the last sentence, maybe in parens (For a complete description of all states, see Authentication API documentation.

File: _source/docs/guides/okta_auth_sdk.md [hunk 6]
```diff
@@ -4,138 +4,171 @@ title: Okta Auth SDK
...
 >
-> * `status` - Matches the transaction's state
-> * `data` - The response body as an object
-
-### SUCCESS
+> * Methods listed for a state are not always guaranteed to be available
+>
+> * `stateToken` is never required as a parameter
+>
+> * All transactions have a `data` attribute that contains the response body as an object
```
Comment (mystiberry-okta) [inline _source/docs/guides/okta_auth_sdk.md]: We should use periods in list items if the item is a sentence. These both are, as is the intro sentence on line 56. (We ignore this rule in tables).

File: _source/docs/guides/okta_auth_sdk.md [hunk 7]
```diff
@@ -4,138 +4,171 @@ title: Okta Auth SDK
...
---------------------- | --------  | -----------
-`expiresAt`           | [Date String](https://tools.ietf.org/html/rfc3339) | Time the sessionToken will expire
-`sessionToken`        | [Session Token](/docs/api/resources/authn.html#session-token) | Token that can be exchanged for a cookie
-`user`                | [User](http://localhost:4000/docs/api/resources/authn.html#user-object) | Subset of user properties
+#### Locked Out
 
-### LOCKED_OUT
+The user account is locked; self-service unlock or admin unlock is required.
 
 Name                  |   Type    | Description
```
Comment (mystiberry-okta) [inline _source/docs/guides/okta_auth_sdk.md]: For all these tables, "Name" should probably be "Property" (sorry!).

File: _source/docs/guides/okta_auth_sdk.md [hunk 8]
```diff
@@ -195,16 +266,27 @@ Name                  |   Type    | Description
 `send`                | Function | Takes a `name` parameter that's usually `sms` or `email`
```
Comment (mystiberry-okta) [inline _source/docs/guides/okta_auth_sdk.md]: Is there a list of valid values somewhere for name? (I couldn't find anything in the other api docs linked to here). Here and elsewhere, I wonder when I see "usually" if there are other possible values.

File: _source/docs/guides/okta_auth_sdk.md [hunk 9]
```diff
@@ -4,138 +4,171 @@ title: Okta Auth SDK
...
 
 ~~~ html
-<script src="OktaAuth.js" type="text/javascript"></script>
+<script src="https://ok1static.oktacdn.com/assets/js/sdk/okta-auth-sdk-1.0.0.min.js" type="text/javascript"></script>
 ~~~
 
 ## Authentication Flow
 
-The goal of an authentication flow is setting an Okta cookie. Simple!
+The goal of an authentication flow is setting an Okta cookie. We use [A+ Promises](https://promisesaplus.com/). If you want an introduction to Promises, [read this](http://documentup.com/kriskowal/q/). Here's an example of how we use Promises:
```
Comment (rchild-okta) [inline _source/docs/guides/okta_auth_sdk.md]: The goal of an authentication flow is to set an Okta session cookie on the user's browser. 

All AuthSDK methods return [Promises](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise), deferred objects that resolve when the method has finished. For more information about why Promises are useful, and how to use our specific promise library, check out  the [Q.js readme](https://github.com/kriskowal/q).

An example auth flow would look like this:

File: _source/docs/guides/okta_auth_sdk.md (hunk 5 above)
Comment (mystiberry-okta) [inline _source/docs/guides/okta_auth_sdk.md]: Yes!

File: _source/docs/guides/okta_auth_sdk.md (hunk 2 above)
Comment (lboyette-okta) [inline _source/docs/guides/okta_auth_sdk.md]: @stsai-okta suggested the change. If we switched to the old version, we'd remove the verb, so we'd need to remove the verb in "Add CORS to your domain in Okta". So, how's this?:

- An Okta org - Register for [Okta Developer Edition](https://www.okta.com/developer/signup/) if you don't have an existing org
- A domain that is CORS enabled for your org

File: _source/docs/guides/okta_auth_sdk.md (hunk 7 above)
Comment (lboyette-okta) [inline _source/docs/guides/okta_auth_sdk.md]: No worries! Will fix

File: _source/docs/guides/okta_auth_sdk.md (hunk 8 above)
Comment (lboyette-okta) [inline _source/docs/guides/okta_auth_sdk.md]: I haven't seen other values, but I didn't want to rule them out. I'll remove "usually", but we'll have to update it once we support other values.

File: _source/docs/guides/okta_auth_sdk.md (hunk 4 above)
Comment (rchild-okta) [inline _source/docs/guides/okta_auth_sdk.md]: Now that I think about it, I'm not sure we need the preceding paragraph. Maybe something like the following instead?

## Transactions

When AuthSDK methods resolve, they return a Transaction object that encapsulates the new state in the authentication flow. This Transaction object contains metadata about the current state, and methods that can be used to progress to the next state.

More information about possible auth states, their definitions, and the flow between them can be found in our [Authentication Api documentation].

### Transaction states
(I think we should just break it out into a separate sub section)

Comment (mystiberry-okta) [inline _source/docs/guides/okta_auth_sdk.md]: are they authentication states or transaction states--we should be consistent. ("auth states"). And I'd remove "possible" as we seldom document impossible states. (just a minimalism thing).

File: _source/docs/guides/okta_auth_sdk.md (hunk 8 above)
Comment (mystiberry-okta) [inline _source/docs/guides/okta_auth_sdk.md]: Okay! You can give me the JIRA when that happens :)

File: _source/docs/guides/okta_auth_sdk.md (hunk 3 above)
Comment (lboyette-okta) [inline _source/docs/guides/okta_auth_sdk.md]: I'll make the script => script tag change. I'll also be more specific with page and say "target web page". The login page may not be the only place they want to use the SDK.

File: _source/docs/guides/okta_auth_sdk.md (hunk 4 above)
Comment (lboyette-okta) [inline _source/docs/guides/okta_auth_sdk.md]: ## Transactions

When Auth SDK methods resolve, they return a Transaction object that encapsulates the new state in the authentication flow. This Transaction object contains metadata about the current state, and methods that can be used to progress to the next state.

More information about Transaction states, their definitions, and the flow between them can be found in our [Authentication API documentation](/docs/api/resources/authn.html#transaction-state).

### Transaction States

File: _source/docs/guides/okta_auth_sdk.md (hunk 6 above)
Comment (lboyette-okta) [inline _source/docs/guides/okta_auth_sdk.md]: will fix!

File: _source/docs/guides/okta_auth_sdk.md (hunk 9 above)
Comment (lboyette-okta) [inline _source/docs/guides/okta_auth_sdk.md]: will change

File: _source/docs/guides/okta_auth_sdk.md (hunk 1 above)
Comment (lboyette-okta) [inline _source/docs/guides/okta_auth_sdk.md]: will do

**Comment by mystiberry-okta** [comment]
Looks great to me!

**Comment by rchild-okta** [comment]
🚀

**Pull request merged**